test/provider_coarse.cpp

```diff
@@ -284,11 +281,10 @@ TEST_P(CoarseWithMemoryStrategyTest, coarseProvider_wrong_params_5) {
 
     const size_t init_buffer_size = 20 * MB;
 
-    // Preallocate some memory
-    std::unique_ptr<char[]> buffer(new char[init_buffer_size]);
-    void *buf = buffer.get();
+    // preallocate some memory and initialize the vector with zeros
+    std::vector<char> buffer(init_buffer_size, 0);
+    void *buf = buffer.data();
```

Comment:
I assume implicit cast by Coverity due to type conversion: 
`data()` returns `value_type*`, in this case `char*`
https://cplusplus.com/reference/vector/vector/data/

Reply:
@KFilipek Done